recipes-multimedia/libcamera/libcamera_0.0.1.bb

```diff
@@ -34,6 +34,8 @@ EXTRA_OEMESON = " \
     -Ddocumentation=disabled \
 "
 
+MESON_BUILDTYPE:printnanny = "release"
```

Comment:
This should use the default `plain` build type.

Reply:
`plain` was failing to build. Fortify is enabled, but it fails the build when it is enabled and optimization is not.

`plain` explicitly sets `optimization` 0; release sets `2`. The other solution would seem to be to disable fortify, but if you know of a third way to solve this I'm happy to do that.

Reply:
```
| FAILED: src/libcamera/base/libcamera-base.so.0.0.1.p/event_dispatcher_poll.cpp.o
| aarch64-bitsy-linux-g++ -mcpu=cortex-a72 -march=armv8-a+crc -fstack-protector-strong -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security --sysroot=/poky/build/tmp/work/cortexa72-bitsy-linux/libcamera/202210+gitAUTOINC+7219110a12-r0/recipe-sysroot -Isrc/libcamera/base/libcamera-base.so.0.0.1.p -Isrc/libcamera/base -I../git/src/libcamera/base -Iinclude -I../git/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Werror -std=c++17 -O0 -Wshadow -include /poky/build/tmp/work/cortexa72-bitsy-linux/libcamera/202210+gitAUTOINC+7219110a12-r0/build/config.h -Wno-unaligned-access -fPIC -pthread -DLIBCAMERA_BASE_PRIVATE -MD -MQ src/libcamera/base/libcamera-base.so.0.0.1.p/event_dispatcher_poll.cpp.o -MF src/libcamera/base/libcamera-base.so.0.0.1.p/event_dispatcher_poll.cpp.o.d -o src/libcamera/base/libcamera-base.so.0.0.1.p/event_dispatcher_poll.cpp.o -c ../git/src/libcamera/base/event_dispatcher_poll.cpp
| In file included from /poky/build/tmp/work/cortexa72-bitsy-linux/libcamera/202210+gitAUTOINC+7219110a12-r0/recipe-sysroot/usr/include/c++/12.2.0/aarch64-bitsy-linux/bits/os_defines.h:39,
|                  from /poky/build/tmp/work/cortexa72-bitsy-linux/libcamera/202210+gitAUTOINC+7219110a12-r0/recipe-sysroot/usr/include/c++/12.2.0/aarch64-bitsy-linux/bits/c++config.h:655,
|                  from /poky/build/tmp/work/cortexa72-bitsy-linux/libcamera/202210+gitAUTOINC+7219110a12-r0/recipe-sysroot/usr/include/c++/12.2.0/bits/stl_algobase.h:59,
|                  from /poky/build/tmp/work/cortexa72-bitsy-linux/libcamera/202210+gitAUTOINC+7219110a12-r0/recipe-sysroot/usr/include/c++/12.2.0/list:60,
|                  from ../git/include/libcamera/base/event_dispatcher_poll.h:10,
|                  from ../git/src/libcamera/base/event_dispatcher_poll.cpp:8:
| /poky/build/tmp/work/cortexa72-bitsy-linux/libcamera/202210+gitAUTOINC+7219110a12-r0/recipe-sysroot/usr/include/features.h:412:4: error: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Werror=cpp]
|   412 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
|       |    ^~~~~~~
| cc1plus: note: unrecognized command-line option '-Wno-unaligned-access' may have been intended to silence earlier diagnostics
| cc1plus: all warnings being treated as errors
| ninja: build stopped: subcommand failed.
| WARNING: exit code 1 from a shell command.
ERROR: Task (/poky/meta-bitsy/recipes-multimedia/libcamera/libcamera_0.0.1.bb:do_compile) failed with exit code '1'
```

Reply:
Can you post full `do_configure` and `do_compile` logs? Here are my configure/compile logs for comparison: 
https://gist.github.com/leigh-johnson/2694d7e5d38ec2576800a9037a346245

Reply:
[log.do_compile.505.txt](https://github.com/bitsy-ai/meta-bitsy/files/10099442/log.do_compile.505.txt)
[log.do_configure.330.txt](https://github.com/bitsy-ai/meta-bitsy/files/10099443/log.do_configure.330.txt)